report/renderer/details-renderer.js

```diff
@@ -376,7 +379,106 @@ export class DetailsRenderer {
+      categoryChipEl.textContent = matchedEntity.category;
+      firstTdEl.append(' ', categoryChipEl);
+    }
+
+    if (matchedEntity.isFirstParty) {
+      const firstPartyChipEl = this._dom.createElement('span');
+      firstPartyChipEl.classList.add('lh-audit__adorn', 'lh-audit__adorn1p');
+      firstPartyChipEl.textContent = Globals.strings.firstPartyChipLabel;
+      firstTdEl.append(' ', firstPartyChipEl);
+    }
+
+    if (matchedEntity.homepage) {
+      const entityLinkEl = this._dom.createElement('a');
+      entityLinkEl.href = matchedEntity.homepage;
+      entityLinkEl.target = '_blank';
+      entityLinkEl.title = Globals.strings.openInANewTabTooltip;
+      entityLinkEl.classList.add('lh-report-icon--external');
+      firstTdEl.append(' ', entityLinkEl);
+    }
+  }
+
+  /**
+   * Renders an aggregated row.
+   * @param {TableItem} item
+   * @param {LH.Audit.Details.TableColumnHeading[]} headings
+   */
+  _renderEntityGroupRow(item, headings) {
+    const entityColumnHeading = {...headings[0]};
+    // In subitem-situations (unused-js), ensure Entity name is not rendered as code, etc.
+    entityColumnHeading.valueType = 'text';
+    const aggregateRowHeadings = [entityColumnHeading, ...headings.slice(1)];
+    const fragment = this._dom.createFragment();
+    fragment.append(this._renderTableRow(item, aggregateRowHeadings));
+    return fragment;
+  }
+
+  /**
+   * Computes entity group items and column totals from TableItem's
+   * @param {TableLike} details
+   * @return {TableItem[]}
+   */
+  _getEntityGroupItems(details) {
+    const {items, headings, sortedBy} = details;
+    // Exclude pre-aggregated audits and results without entity classification.
+    // Eg. Third-party Summary comes pre-aggregated.
+    if (!items.length || details.isEntityGrouped || !items.some(item => item.entity)) {
+      return [];
+    }
+    const skippedColumns = new Set(details.skipSumming || []);
+    const supportedAggregations = ['bytes', 'numeric', 'ms', 'timespanMs'];
+    /** @type {string[]} */
+    const aggregateKeys = [];
+    for (const heading of headings) {
+      if (!heading.key || skippedColumns.has(heading.key)) continue;
+      if ('valueType' in heading && supportedAggregations.includes(heading.valueType)) {
+        aggregateKeys.push(heading.key);
+      }
+    }
+
+    // Grab the first column's key to aggregate by entity
+    const primaryKey = headings[0].key;
+    if (!primaryKey) return [];
+
+    /** @type {Map<string | undefined, TableItem>} */
+    const byEntity = new Map();
+    for (const item of items) {
+      const entityName = typeof item.entity === 'string' ? item.entity : undefined;
+      const aggregate = byEntity.get(entityName) || {
+        [primaryKey]: entityName || Globals.strings.unattributable,
+        entity: entityName,
+      };
+      for (const key of aggregateKeys) {
+        aggregate[key] = Number(aggregate[key] || 0) + Number(item[key]);
```

Comment:
Not sure if this would ever happen, but I think it makes sense to be defensive here. If `item[key]` is undefined for some reason this is `NaN`.

```suggestion
        aggregate[key] = Number(aggregate[key] || 0) + Number(item[key] || 0);
```